Feature/votings info in modals

zyngielg opened the pull request:
![image](https://user-images.githubusercontent.com/29864544/50738588-ed647200-11d5-11e9-94f2-042526a53416.png)
![image](https://user-images.githubusercontent.com/29864544/50738609-23a1f180-11d6-11e9-9c6e-bc4ca1f6844d.png)

akwodkiewicz commented:
I suggest using less words in Voting info. For example:

**Voting deadline**
January 8th 2019, 4:00:00 PM

Also, let's use _Results expiry date_ uniformly in whole application.
This is my suggestion snippet for `VoteModal.tsx`

```jsx
<Panel.Body>
  <ControlLabel style={{ fontSize: "1.2em" }}>Question</ControlLabel>
  <p>{this.props.voting.info.question}</p>
  <ControlLabel style={{ fontSize: "1.2em" }}>Voting deadline</ControlLabel>
  <p>{covertTimestampToDate(this.props.voting.info.votingEndTime)}</p>
  <ControlLabel style={{ fontSize: "1.2em" }}>Results expiry date</ControlLabel>
  <p>{covertTimestampToDate(this.props.voting.info.resultsEndTime)}</p>
  <ControlLabel style={{ fontSize: "1.2em" }}>Voting type</ControlLabel>
  <p>{this.props.voting.info.isPrivate ? "Private" : "Public"}</p>
<Panel.Body>
```

zyngielg commented:
- fixed HomePage resizing bug
![image](https://user-images.githubusercontent.com/29864544/50741262-f3207e80-11fa-11e9-8d96-2af339b3de5b.png)
- changed displayed info format according to the PR review
- added table with results 
- removed 0% data from piechart

akwodkiewicz commented:
:+1: for HomePage resize bug, I forgot to mention it earlier

akwodkiewicz commented on client/src/components/vote/ResultsModal.tsx:
```diff
@@ -15,39 +16,89 @@ export default class ResultsModal extends Component<IResultsModalProps> {
+
   public render() {
+    const dic = this.getSortedAnswersWithResults();
+    console.log(dic);
```
Delete console log

akwodkiewicz commented on client/src/components/vote/VoteModal.tsx:
```diff
@@ -31,53 +34,95 @@ export default class VoteModal extends Component<IVoteModalProps, IVoteModalStat
+                  <ControlLabel style={{ fontSize: "1.2em" }}>Question</ControlLabel>
+                  <p>{this.props.voting.info.question}</p>
+                  <ControlLabel style={{ fontSize: "1.2em" }}>Voting deadline</ControlLabel>
+                  <p>Voting possible until {covertTimestampToDate(this.props.voting.info.votingEndTime)}</p>
```
```suggestion
                  <p>{covertTimestampToDate(this.props.voting.info.votingEndTime)}</p>
```
display plain date as voting deadline info

akwodkiewicz replied:
This way it's unified with `<ResultsModal />`

akwodkiewicz commented on client/src/components/vote/ResultsModal.tsx:
```diff
@@ -15,39 +16,89 @@ export default class ResultsModal extends Component<IResultsModalProps> {
+                    <tr>
+                      <td style={{ width: "65%" }}>{element.answer}</td>
+                      <td style={{ textAlign: "center" }}>{element.result}</td>
+                      <td style={{ textAlign: "center" }}>{((100 * element.result) / dic.length).toFixed(2)}%</td>
```
That's not right

![percentages](https://user-images.githubusercontent.com/22861194/50743029-ec056a80-1212-11e9-9cbd-efed62735ad2.png)

You should divide by sum of all votes (here: 2), not by number of options (6). So it should be like this:
`dic.reduce((partial, obj) => partial + obj.result, 0)`

But I'm not creating a one-click suggestion, because it would be more efficient to evaluate it 1 time (at the top of `render`), not n-times.

akwodkiewicz commented on client/src/components/vote/ResultsModal.tsx:
```diff
@@ -15,39 +16,89 @@ export default class ResultsModal extends Component<IResultsModalProps> {
+
+    results.sort().reverse();
+    const votingInfo: VotingInfo = this.props.voting.info;
+    const dic = [];
```
```suggestion
    const dic: Array<{ answer: string; result: number }> = [];
```
add function return type (implicitly)

akwodkiewicz replied:
It's easier to write `map/reduce` functions

akwodkiewicz commented on client/src/components/vote/ResultsModal.tsx:
```diff
@@ -15,39 +16,89 @@ export default class ResultsModal extends Component<IResultsModalProps> {
+  };
+
   public render() {
+    const dic = this.getSortedAnswersWithResults();
```
Hmmmmm :thinking: :thought_balloon:

`dic.length`... `dic.reduce`...

akwodkiewicz replied:
:guardsman: I find your variable name choices disturbing :eyes:

zyngielg replied:
XD

akwodkiewicz commented:
Extracting suggestion to separate comment:
Can you add voting address both to `<VoteModal />` and `<ResultsModal />`?

zyngielg commented:
The force push renamed "tx" to "voting" in last commit heh
